locale.c

```diff
@@ -3823,21 +3823,61 @@ Perl_warn_problematic_locale()
 #  endif /* USE_LOCALE_CTYPE */
 
 STATIC void
-S_new_LC_ALL(pTHX_ const char *unused, bool force)
+
+#  ifdef LC_ALL
+
+S_new_LC_ALL(pTHX_ const char *lc_all, bool force)
+
+#  else
+
+S_new_LC_ALL(pTHX_ const char ** individ_locales, bool force)
+
+#  endif
+
 {
     PERL_ARGS_ASSERT_NEW_LC_ALL;
-    PERL_UNUSED_ARG(unused);
 
     /* new_LC_ALL() updates all the things we care about.  Note that this is
      * called just after a change, so uses the actual underlying locale just
      * set, and not the nominal one (should they differ, as they may in
      * LC_NUMERIC). */
 
+#  ifdef LC_ALL
+
+    const char * individ_locales[LC_ALL_INDEX_] = { NULL };
+
+    switch (parse_LC_ALL_string(lc_all,
+                                (const char **) &individ_locales,
```

Comment:
The `(const char **) &individ_locales` can just be `individ_locales` due to the way arrays decay into pointers.